uvm/hier/usr/lib/python3/dist-packages/tests/test_peformance.py

```diff
@@ -0,0 +1,234 @@
+                if application_name != "router":
+                    application = uvmContext.appManager().instantiate(application_name, default_policy_id)
+
+                ## Loop through app to get average throughput
+                application_throughput = []
+                for i in range(Performance_throughput_iterations):
+                    iperf_results = global_functions.get_iperf_results(duration=Performance_throughput_duration)
+                    application_throughput.append(iperf_results[1])
+                application_average = sum(application_throughput) / len(application_throughput)
+
+                if application != None:
+                    uvmContext.appManager().destroy( application.getAppSettings()["id"] )
+                    application = None
+
+                results[tcp_buffer_size][application_name] = {
+                    "throughput": application_average,
+                }
+
+        ##
+        ## Report
+        ## 
+        print("=" * 40)
+        print("Summary:")
+        print(f"average bypass throughput: {global_functions.to_si_prefix(bypass_throughput_average)}")
+        print("By size:")
+        for size in results:
+            for application_name in results[size]:
+                current_throughput = results[size][application_name]["throughput"]
+                print(f"{size:6} {application_name:20}: {global_functions.to_si_prefix(current_throughput)}, of bypass {(current_throughput / bypass_throughput_average) * 100:.2f}%")
+
+        print("By size, relative to last size:")
+        last_size = None
+        for size in results:
+            for application_name in results[size]:
+                current_throughput = results[size][application_name]["throughput"]
+                if last_size is not None:
+                    last_throughput = results[last_size][application_name]["throughput"]
+                    print(f"{size:6} vs {last_size:6}: {application_name:20}: {global_functions.to_si_prefix(current_throughput)} vs {global_functions.to_si_prefix(last_throughput)} >? {current_throughput > last_throughput}, diff-t={global_functions.to_si_prefix(abs(current_throughput-last_throughput))}, diff-p={(abs(current_throughput-last_throughput)/current_throughput) * 100:.2f}%")
+
+            last_size = size
+        print("=" * 40)
+
+        ##
+        ## Restore orginal untangle-vm.conf
+        shutil.copyfile(untangle_vm_conf_original_filename, untangle_vm_conf_filename)
+        uvmContext = global_functions.restart_uvm()
+
+        ##
+        ## Stop iperf server
+        ##
+        iperf_running = 0
+        timeout = 60
+        while iperf_running == 0 and timeout > 0:
+            timeout -= 1
+            remote_control.run_command("pkill --signal 9 iperf", host=global_functions.iperf_server)
+            time.sleep(1)
+            iperf_running = remote_control.run_command("pidof iperf", host=global_functions.iperf_server)
+
+        assert(True is True)
```

Comment:
I assume this is here for figuring things out and should probably be removed?

Reply:
Thanks for catching that.